akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -370,6 +372,7 @@ private[stream] final class VirtualProcessor[T] extends Processor[T, T] {
           try {
             subscriptionStatus.set(s)
             tryOnSubscribe(s, sub)
+            if (isCancelled) throw canNotSubscribeTheSameSubscriberMultipleTimesException
```

Comment:
Note that the tracking of additional subscribers is on a best effort basis as the rule for no concurrent signalling still applies. But yes, @ktoso, I agree with your sentiment! :)